prive/utils/data.py

```diff
@@ -127,4 +127,38 @@ def index_split(max_index, split_size, num_splits):
     np.random.shuffle(index_array)
     indices += [index_array[split_size*i : split_size*(i+1)] for i in range(remainder_splits)]
 
-    return indices
\ No newline at end of file
+    return indices
+
+def get_dtype(representation, column_type):
+    """
+
+    Based on the json file schema from the dataset module map type/representation to pandas types.
+    See data-schema-doc.rst documentation for more details.
+
+    Parameters
+    ----------
+    representation: object
+        Representation of column based on the schema this can be a string, and integer, or a list
+    column_type: str
+        String referring to abstract type of the data column (e.g. countable/ordered, finite, etc).
+
+    Returns
+    -------
+    dtype
+        A type for the given column.
+
+    """
+    if representation == 'integer':
+        return np.int
+    elif representation == 'number':
+        return np.float
+    elif representation == 'date':
+        return pd.datetime
+    elif isinstance(representation, (list, str)):
+        ordered = True if 'ordered' in column_type else False
+        return pd.CategoricalDtype(categories=representation, ordered=ordered)
+    elif isinstance(representation, (int)):
+        ordered = True if 'ordered' in column_type else False
+        return pd.CategoricalDtype(categories=range(representation), ordered=ordered)
+    else:
+        np.object
```

Comment:
I'm a little worried about the catch-all here. If we're the ones writing the schema requirements, we should be careful not to allow people to not follow it. I think I would prefer a comprehensive set of `elif` until all possible schema values have been done, and then the final `else:` raises a `Warning`/`Error`.